lib/logstash/outputs/s3.rb

```diff
@@ -84,6 +86,9 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   # S3 bucket
   config :bucket, :validate => :string
 
+  # endpoint
+  config :endpoint, :validate => :string
```

Comment:
region: it may be valid to provide both a region and an endpoint for some providers.  v4 signing includes the region in its signature.

docs: added blurb and example.